src/nspawn/nspawn.c

```diff
@@ -2493,6 +2510,65 @@ static int outer_child(
         if (r < 0)
                 return r;
 
+        if (arg_pivot_sysroot) {
```

Comment:
can you split this out into its own function please? (and place that in nspawn-mount.c next to the volatile stuff)